src/services/TestResultsService.ts

```diff
@@ -232,252 +267,29 @@ export class TestResultsService {
-      payload.testTypes = testTypesWithTestCodesAndClassification;
-      const payloadWithTestNumber = await this.setTestNumber(payload);
-      const payloadWithExpiryDate = await this.generateExpiryDate(payloadWithTestNumber);
-      const payloadWithCertificateNumber = this.generateCertificateNumber(payloadWithExpiryDate);
-      const payloadWithAnniversaryDate = this.calculateAnniversaryDate(payloadWithCertificateNumber);
-      const payloadWithVehicleId = this.setVehicleIdToVRM(payloadWithAnniversaryDate);
-      return await this.testResultsDAO.createSingle(payloadWithVehicleId);
-    } catch (error) {
-      if (error.statusCode === 400 && error.message === enums.MESSAGES.CONDITIONAL_REQUEST_FAILED) {
-        console.log("Error in insertTestResult > getTestTypesWithTestCodesAndClassification: Test Result id already exists", error);
-        return Promise.reject(new models.HTTPResponse(201, enums.MESSAGES.ID_ALREADY_EXISTS));
-      }
-      console.log("Error in insertTestResult > getTestTypesWithTestCodesAndClassification", error);
-      return Promise.reject(new models.HTTPError(500, enums.MESSAGES.INTERNAL_SERVER_ERROR));
-    }
-  }
-
-  public async setTestNumber(payload: models.ITestResultPayload) {
-    const promiseArray: any[] = [];
-    if (payload.testTypes) {
-      payload.testTypes.forEach((testType: { testNumber: any; }) => {
-        const promise = this.testResultsDAO.getTestNumber()
-          .then((testNumberResponse: { testNumber: any; }) => {
-            testType.testNumber = testNumberResponse.testNumber;
-          });
-
-        promiseArray.push(promise);
-      });
-      await Promise.all(promiseArray);
-      return payload;
-    } else {
-      return Promise.resolve(payload);
-    }
-  }
-
-  public reasonForAbandoningPresentOnAllAbandonedTests(payload: models.ITestResultPayload) {
-    let bool = true;
-    if (payload.testTypes) {
-      if (payload.testTypes.length > 0) {
-        payload.testTypes.forEach((testType: { testResult: string; reasonForAbandoning: any; }) => {
-          if (testType.testResult === enums.TEST_RESULT.ABANDONED && !testType.reasonForAbandoning) {
-            bool = false;
-          }
-        });
-      }
-    }
-    return bool;
-  }
-
-  /**
-   * Note: When performing actions on a moment instance, it gets mutated
-   * Note: Expiry dates on the payload should be set at the UTC start of day.
-   *
-   * @param payload
-   */
-  public async generateExpiryDate(payload: models.ITestResultPayload): Promise<models.ITestResultPayload> {
-    moment.tz.setDefault("UTC");
+    // FIXME: Suggestion to remove this line as it is duplicate of if(!payload) in postTestResults.ts
+    // if (Object.keys(payload).length === 0) { // if empty object
+    //   return Promise.reject(new models.HTTPError(400, enums.ERRORS.PayloadCannotBeEmpty));
+    // }
```

Comment:
If we are removing code, remove it, rather than comment it out.

Reply:
already removed 18783.